Modules/DataCollection/classes/class.ilDataCollectionDataSet.php

```diff
@@ -113,6 +113,9 @@ public function importRecord(
         ilImportMapping $a_mapping,
         string $a_schema_version
     ): void {
+        foreach ($a_rec as &$value) {
```

Comment:
Hi @iszmais , I suggest to use `\ilDataSet::stripTags` for input filtering (like in all other `DataSet` based XML imports) and known mechanisms for escpaing/encoding when putting user input into HTML templates.

Reply:
@mjansenDatabay Thank you for your feedback!

I oppose the usage of strip tags here for two reasons:
- This would limit the presentation of titles. Especially within a learning management system, I see that as a concern. This would make it hard to create an Object regarding HTML for educational purposes (e.g. my DataCollection "Cool ways to use the <div> Tag!" would then only be shown as "Cool ways to use the Tag!" which would be confusing)
- This would make the approach disjunct from the "manual" creation, which currently allows you to save and present tags within the title (even when there are slightly modified).

I may take this concern to a future approach to stricten the input of title, but those should then apply to all ways of creation.

Greetings,
@iszmais

Reply:
Hi @iszmais ,

I would like to add two thoughts:

1. IMO the input handling in the XML import process should be similar/identical to the input handling when users submit data via the ILIAS graphical user interface (GUI). It is just another application-endpoint. So applying `\ilDataSet::stripTags` would be the most appropriate solution for the problem here.
2. The `title` is a property of the `Object` component, **not** of the `DataCollection` component. Although the title is not handled centrally by the `Object` component in imports (at the moment), the same business logic (restrictions etc.) applied by the `Object` component when handling user input via the GUI **must** apply here. So IMO the "limited presentation of title"-argument is not valid here.

Reply:
Thank you for your response @mjansenDatabay ,

it seems to be a bit of a misunderstanding here:
The current common state is that tags are **not** stripped via GUI. They are preserved in a readable form.

I'm not totally if this applies to all types and how the abstract object handles it, but I checked this for the most common types like course, group, category etc. 

Therefore, I support your point, but that resolves into preserved tags via import as well.

Greetings,
@iszmais